bot/controllers/verify.js

```diff
@@ -0,0 +1,152 @@
+        // Method to generate random numeric verification code
+        // Modified to fit style guide from this SO answer:
+        // https://stackoverflow.com/a/39774334
+        const generateCode = (n) => {
+          const add = 1;
+          let max = 12 - add;
+          let min = 0;
+          if (n > max) {
+            return generateCode(max) + generateCode(n - max);
+          }
+          max = 10000000;
+          min = max / 10;
+          const number = Math.floor(Math.random() * (max - (min + 1))) + min;
+          util.log(number);
+          return ('' + number).substring(add);
+        };
+        // We can set `codeLength` to whatever length we want the verif code to be.
+        // Recommend ngt 8 digits.
+        if (validDomains.includes(emailDomain)) {
+          const codeLength = 6;
+          const code = generateCode(codeLength);
+          // TODO: Set `time` prop to 600000 (10min)
+          const collector = message.channel.createMessageCollector(
+            m => m.content.includes(code),
+            // 15000ms only for testing!!!
+            { time: 150000 });
+          collector.on('collect', (m) => {
+            util.log('Collected', m.content, 3);
+            models.Member.create({
+              discorduser: m.author.username,
+              email,
+              uuid: uuidv4(),
+              verified: 1,
+            }).then(util.log).catch(util.error);
+
+            // TODO: Database integration/email verif process
+            message.reply('Thanks! I\'ll get to work adding you the servers right away!');
+          });
+          collector.on('end', (collected) => {
+            util.log('Items', collected.size, 3);
+            if (collected.size === 0) {
+              // TODO: ping admin team on verification fail
+              message.reply('Uh-oh, it looks like you weren\'t able to get the right verification code back to me in time. I\'ve contacted the Armada admins so we can get this straightened out right away.');
+            }
+          });
+          // Set up Nodemailer to send emails through gmail
+          const sendVerifyCode = nodemailer.createTransport({
+            service: 'gmail',
+            auth: {
+              user: process.env.EMAIL_USERNAME,
+              pass: process.env.EMAIL_PASS,
+            },
+          });
+          // Nodemailer email recipient & message
+          // TODO: Build email template
+          const mailOptions = {
+            from: 'discord.maxbot@gmail.com',
+            to: email,
+            subject: 'Armada Verification Code',
+            html: `<p>Verification Code: ${code}</p>`,
+          };
+          // Call sendMail on sendVerifyCode
+          // Pass mailOptions & callback function
+          sendVerifyCode.sendMail(mailOptions, (err, info) => {
+            if (err) {
+              message.reply('Oops, looks like the email can not be sent. Its not you, its me.. Please contact a moderator and let them know I have failed.');
```

Comment:
Messages should be replying with variables defined at the top of the action callback for more comprehensible future editing of the messages.

Reply:
Thank you, message has been stored in variable to resolve this issue.